control-plane/pkg/reconciler/source/source.go

```diff
@@ -144,15 +147,40 @@ func (r Reconciler) reconcileConsumerGroup(ctx context.Context, ks *sources.Kafk
 			Extensions: ks.Spec.CloudEventOverrides.Extensions,
 		}
 	}
+
 	if kt, ok := ks.Labels[sources.KafkaKeyTypeLabel]; ok && len(kt) > 0 {
 		expectedCg.Spec.Template.Spec.Configs.KeyType = &kt
 	}
 
+	// TODO: make these parameters below configurable and maybe unexposed
+	if ks.Annotations != nil {
```

Comment:
Below to detect updates we compare only `cg.Spec` and `ks.Spec` [1], given we're propagating specific annotations as well, I think we should include annotations as well.

`equality.Semantic.DeepDerivative(expectedCg.Spec, cg.Spec)`

maybe a unit test for this scenario is helpful.

Reply:
Good catch. Will add

Reply:
Covered by this test: https://github.com/knative-sandbox/eventing-kafka-broker/pull/2608/files#diff-1dd13376cedd8412c716fe91677e1fbfa62c0f026b9cecebf8e33319cdd844bdR685